docs/codesigning/common-issues.md

```diff
@@ -32,8 +32,121 @@ Your project defines a provisioning profile which doesn't exist on your local ma
+
+In my case, it was because my export was built without bitcode, but I had static lib frameworks that were built with it (inside `IDEDistribution.standard.log` within the `.xcdistributionlogs` file):
+
+>2017-03-31 16:49:06 +0000  Validating IPA structure...
+>2017-03-31 16:49:06 +0000  error: Failed to verify bitcode in opus.framework/opus:
+error: Bundle only contains bitcode-marker /var/folders/s1/0h7d0scx6xqdthft328yfk0m0000gn/T/IDEDistributionThinningStep.KRt/Payload/Viv.app/Frameworks/opus.framework/opus (armv7)
+
+I rebuilt that framework without bitcode and it worked like a charm! 😅 
+
+### Run commands yourself
+
+_gym_ does a great job of showing what commands it's going to run. Inspect the files that it generates and run these commands yourself, as many times as you need to, in order to isolate the problems.
+
+Example, here's the export command that _gym_ shows:
+
+>/usr/bin/xcrun /Users/rob/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.24.0/gym/lib/assets/wrap_xcodebuild/xcbuild-safe.sh -exportArchive -exportOptionsPlist '/var/folders/s1/0h7d0scx6xqdthft328yfk0m0000gn/T/gym_config20170330-65021-1xbw0ov.plist' -archivePath /Users/rob/Library/Developer/Xcode/Archives/2017-03-30/MyApp\ 2017-03-30\ 21.12.02.xcarchive -exportPath '/var/folders/s1/0h7d0scx6xqdthft328yfk0m0000gn/T/gym_output20170330-65021-u80ojn'
+
+From that, I can see which `exportOptionsPlist` it's using, which has this in it:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>method</key>
+	<string>development</string>
+</dict>
+</plist>
+```
+
+And I can see which `.xcarchive` it's using to check if it built that with the correct provisioning profile.
+
+Most importantly, I can keep running that entire command until I figure out why it's breaking (e.g. try installing these gems, try using system Ruby).
+
+### Other Causes
+
+Keep in mind, this list could be infinite since this `No applicable devices found` error just manifests itself from underlying errors. Your best bet is to dig into the logs and see which error is actually being thrown.
+
+- Using the wrong `export_method` or some other error in the generated `exportPlist` that gets generated
+- Root Apple WWDR certificate had expired; check the Keychain for expired certificates
+- Env vars not being unset, [relevant PR](https://github.com/fastlane-old/gym/pull/176)
+- Including bitcode sometimes caused it (can use `include_bitcode: false`) in your _gym_ command and check if all frameworks match your bitcode requirements
+- Provisioning profiles not matching capabilities and beta-reports-active flag on .entitlements file.
```

Comment:
Isn't another error case that the gems you listed above are not installed on the system ruby? So running `gem install [gem_names]` for the system ruby should solve it, right? `rvm use system && gem install [gem_names]`